### PR TITLE
check for java bytecode index in the existing coverage

### PR DESCRIPTION
**Pull request by lucasccordeiro**
check whether the java bytecode index matches for the existing coverage goals/

### CODE REVIEW

File: src/goto-instrument/cover.cpp [hunk 1]
```diff
@@ -218,10 +218,14 @@ bool coverage_goalst::is_existing_goal(source_locationt source_location)
   std::vector<source_locationt>::iterator it = existing_goals.begin();
   while(it!=existing_goals.end())
   {
-    if(!source_location.get_file().compare(it->get_file()) &&
-       !source_location.get_function().compare(it->get_function()) &&
-       !source_location.get_line().compare(it->get_line()))
-         break;
+    if(!source_location.get_java_bytecode_index().compare(
+      it->get_java_bytecode_index()) &&
+      !source_location.get_file().compare(it->get_file()) &&
```
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: `get_java_bytecode_index` should only be compared when using a Java analysis

Comment (lucasccordeiro) [inline src/goto-instrument/cover.cpp]: Now we check whether get_java_bytecode_index is empty.

File: src/goto-instrument/cover.cpp [hunk 2]
```diff
@@ -203,6 +203,32 @@ void coverage_goalst::set_goals(source_locationt goal)
...
+\*******************************************************************/
+
+bool has_common_location(
+source_locationt given_location,
+std::vector<source_locationt>::iterator &known_location)
+{
+  if(!given_location.get_file().compare(known_location->get_file()) &&
+    !given_location.get_function().compare(known_location->get_function()) &&
+    !given_location.get_line().compare(known_location->get_line()))
+  {
```
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: as compare returns an `int`, could you test on `0` here?

File: src/goto-instrument/cover.cpp [hunk 3]
```diff
@@ -218,10 +244,14 @@ bool coverage_goalst::is_existing_goal(source_locationt source_location)
...
-    if(!source_location.get_file().compare(it->get_file()) &&
-       !source_location.get_function().compare(it->get_function()) &&
-       !source_location.get_line().compare(it->get_line()))
-         break;
+    if(has_common_location(source_location, it))
+    {
+      if(source_location.get_java_bytecode_index().empty())
+        break;
+      else if(!source_location.get_java_bytecode_index().compare(
+        it->get_java_bytecode_index()))
```
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: as above

**Comment by mgudemann** [comment]
there is also compile failures on travis, please fix those, too

**Comment by lucasccordeiro** [comment]
Due to several merge conflicts, I have created another PR to handle this issue: https://github.com/diffblue/cbmc/pull/772

**Pull request closed without merging**